lib/remote.provider.js

```diff
@@ -1,7 +1,120 @@
 'use strict';
 
+const jstp = require('@metarhia/jstp');
+
 const { StorageProvider } = require('./provider');
+const { RemoteCursor } = require('./remote.cursor');
+
+class RemoteProvider extends StorageProvider {
+  constructor() {
+    super();
+    this.connection = null;
+  }
+
+  // Open RemoteProvider
+  //    options <Object> options for jstp connection
+  //      transport <string> jstp transport name
+  //      connectionArgs <Array> arguments to be passed to corresponding
+  //          transport's connect method
+  //    callback <Function>
+  //      error <Error> | <null>
+  //      provider <StorageProvider>
+  open(options, callback) {
+    jstp[options.transport].connect(
+      ...options.connectionArgs,
+      (error, connection) => {
+        if (error) {
+          callback(error);
+          return;
+        }
+        this.connection = connection;
+        callback(null, this);
+      }
+    );
+  }
+
+  // Close RemoteProvider
+  //    callback <Function>
+  //      err <Error> | <null>
+  close(callback) {
+    if (this.connection) {
+      this.connection.close();
+      this.connection = null;
+    }
+
+    process.nextTick(callback);
+  }
+
+  // Get record from GlobalStorage
+  //   id <string> globally unique record id
+  //   callback <Function>
+  //     error <Error> | <null>
+  //     record <Object>
+  get(id, callback) {
+    this.connection.callMethod('provider', 'get', [id], callback);
+  }
+
+  // Set record in GlobalStorage
+  //   record <Object> record to be stored
+  //   callback <Function>
+  //     error <Error> | <null>
+  set(record, callback) {
+    this.connection.callMethod('provider', 'set', [record], callback);
```

Comment:
Maybe check here that `record` has `Id` and return error right away if it doesn't?